Remove ICD server sources from chip_data_model.cmake

adigie opened the pull request:
#### Changes
* Remove ICD server sources from `chip_data_model.cmake`. They are built by `gn`, there is no need to compile them twice.
* Don't depend on ICD sources if it's not enabled.

#### Testing
Tested by building samples from nRF Connected SDK.

adigie committed:
Remove ICD server sources from chip_data_model.cmake

ICD server sources are built by gn, there is no need to compile them
twice.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

adigie committed:
Fix ICD dependencies

Don't depend on ICD if it's not enabled.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

adigie committed:
Fix gn include warnings

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

bzbarsky-apple commented on examples/light-switch-app/cc13x4_26x4/src/AppTask.cpp:
```diff
@@ -62,8 +62,8 @@
-#include "app/icd/server/ICDNotifier.h"
-#endif
+#include "app/icd/server/ICDNotifier.h" // nogncheck
+#end
```
```suggestion
#endif
```

adigie replied:
Fixed, thanks!

bzbarsky-apple commented on src/include/platform/CHIPDeviceConfig.h:
```diff
@@ -41,6 +39,8 @@
 #include CHIP_DEVICE_PROJECT_CONFIG_INCLUDE
 #endif
 
+#include <lib/core/CHIPConfig.h>
```
Makes sense, but this change to how core config stuff works should probably be a separate PR with a clear description and commit message explaining the change, to aid in bisecting if problems arise.

adigie replied:
Opened a separate PR: #37550

adigie replied:
It seems that this change won't be needed as the issue can be fixed by defining `chip_project_config_include` as it's done on other platforms. On `darwin` only `chip_device_project_config_include` was defined and that caused wrong order of the headers.

adigie committed:
iOS TVCasting app: Add missing chip_project_config_include

`chip_project_config_include` has to be set to get correct include order
of configuration files in `CHIPDeviceConfig.h` and prevent macro
redefined warnings coming from overriding defines already defined by
`CHIPConfig.h`.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>
